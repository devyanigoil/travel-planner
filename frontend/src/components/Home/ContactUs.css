/* Body Styling */
body {
    background-color: #f8f9fa; /* Light background */
    font-family: Arial, sans-serif;
  }
  
  /* Contact Section Styling */
  .contact-section {
    background-color: #fff; /* White background for the section */
    padding: 2rem;
    border-radius: 8px;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
    display: flex;
    justify-content: center;
    align-items: flex-start;
    margin: 2rem;
  }
  
  .contact-header {
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    height: 100%;
    padding-top: 1.5rem;
  }
  
  .contact-header h1 {
    font-size: 2.5rem;
    font-weight: bold;
    padding-left: 2rem;
    padding-top: 1rem;
    margin: 0;
  }
  
  .contact-header p {
    font-size: 1rem;
    color: #555;
    padding-left: 2rem;
    padding-top: 1rem;
    margin: 0;
  }
  
  .form-label {
    font-weight: bold;
    font-size: 0.9rem;
  }
  
  .form-control {
    border-radius: 20px;
    border: 1px solid #333;
  }
  
  .btn-submit {
    background-color: #001f54; /* Navy button */
    color: #fff;
    border-radius: 25px;
    font-weight: bold;
    padding: 0.5rem 2rem;
    border: none;
  }
  
  .btn-submit:hover {
    background-color: #003580;
  }
  
  /* Ensure consistent layout with Features
  .container {
    max-width: 1200px;
    margin: 0 auto;
  } */
  
  .row {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }
  
  .col-md-6 {
    flex: 1;
    padding: 1rem;
  }
  
  .contact-us {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 2rem;
  }