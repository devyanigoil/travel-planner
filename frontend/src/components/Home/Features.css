.features {
    padding: 0px !important;
    margin: 0px !important;
    margin-top: 5rem !important;
    display: grid;
    grid-template-columns: 1fr; /* Change to single column for small screens */
    grid-gap: 2rem;
    background-color: #589ea8;
  }
  
  /* Adjust layout on larger screens */
  @media (min-width: 768px) {
    .features {
      grid-template-columns: 1fr 1fr; /* 2 columns on larger screens */
    }
  }
  
  /* Left Div Adjustments */
  .left-div {
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    height: 100%;
    /* margin-left: 1.5rem; Adjust left margin */
    padding-top: 1.5rem; /* Space from top */
  }
  
  .left-div h1 {
    font-size: 2.5rem; /* Adjust font size */
    font-weight: bold;
    color: white;
    padding-left: 2rem; /* Space from left */
    padding-top: 1rem; /* Space from top */
    margin: 0;
  }
  
  /* Right Div Adjustments */
  .right-div {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    padding-right: 2rem;
  }
  
  /* Accordion Customizations */
  .custom-accordion-item {
    border-top: 1px solid black !important;
    border-left: 0 !important;
    border-right: 0 !important;
    background-color: #589ea8 !important;
  }
  
  .custom-accordion-header .accordion-button {
    background-color: #589ea8 !important;
    color: black !important;
    border: none;
  }
  
  .custom-accordion-header .accordion-button:hover {
    background-color: #589ea8 !important;
    color: #000 !important;
  }
  
  .custom-accordion-button {
    font-size: 1.5rem !important;
    padding-bottom: 1.5rem !important;
    padding-top: 1.5rem !important;
    width: 40rem !important;
  }
  