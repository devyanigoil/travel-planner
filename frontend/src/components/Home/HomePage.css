.home-container {
  font-family: "Arial", sans-serif;
  /* color: #333; */
  padding-top: 20px;
  padding-bottom: 20px;
  /* max-width: 1200px; */
  margin: 0 auto;
}

.home-page-img {
  height: 32rem;
  width: 42rem;
}

.header-section {
  text-align: center;
  margin-bottom: 20px;
  /* background: rgb(17, 50, 103); */
  border-radius: 1rem !important;
  /* padding: 1rem; */
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding-left: 0.5rem !important;
  padding-left: 0.5rem !important;
}

.flex-right {
text-align: left !important ; 
padding-left: 1rem; /* Aligns items to the right */
}


.header-section h1 {
  font-size: 2.8rem;
  /* color: #99bfff; Theme yellow for a bright, welcoming effect */
  font-weight: 700;
  margin-bottom: 15px;
  letter-spacing: 1px;
  /* margin-left: -20px; */
}

.header-section h3 {
  font-size: 1.75rem;
  color: #333; /* Darker neutral color to complement the yellow */
  margin-bottom: 20px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.5px;
}

.header-section p {
  font-size: 1.1rem;
  color: #666; /* Subtle gray for a softer feel */
  line-height: 1.6;
  /* max-width: 800px; */
  /* margin: 0 auto 10px; */
  transition: color 0.3s ease;
  max-width: 35rem; /* Restrict the maximum width */
  line-height: 1.6; /* Optional: Improve readability */
  /* margin: 0 auto; */
}

.header-section p:hover {
  color: #99bfff; /* Theme yellow on hover for consistency */
}

.how-it-works {
  margin-bottom: 40px;
}

.how-it-works h2 {
  font-size: 2rem;
  text-align: center;
  margin-bottom: 20px;
  font-weight: bold;
}

.steps {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.step {
  background-color: #99bfff;
  padding: 20px;
  border-radius: 12px;
  width: 30%;
  margin-bottom: 20px;
  text-align: center;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transition: transform 0.3s ease, box-shadow 0.3s ease,
    background-color 0.3s ease;
  position: relative;
  overflow: hidden;
}

.step::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: linear-gradient(135deg, #ffdc80, #99bfff);
  z-index: -1;
  opacity: 0;
  transition: opacity 0.3s ease;
  border-radius: inherit;
}

.step:hover::before {
  opacity: 1;
}

.step:hover {
  transform: scale(1.05);
  box-shadow: 0 8px 20px rgba(0, 0, 0, 0.15);
  background-color: #ffe680;
}

.step h3 {
  font-size: 1.5rem;
  color: black;
  margin-bottom: 10px;
  transition: color 0.3s ease;
}

.step:hover h3 {
  color: #333;
}

.step p {
  font-size: 1rem;
  color: #555;
  transition: color 0.3s ease;
}

.step:hover p {
  color: #333;
}

.step svg {
  color: black; /* Theme color for icons */
  margin-bottom: 10px;
  font-size: 1.5rem;
}

.step:hover svg {
  color: white; /* Slightly darker shade on hover */
}

.step h3 {
  margin-top: 0;
}

.features-section {
  margin-bottom: 40px;
}

.features-section h2 {
  font-size: 2rem;
  text-align: center;
  margin-bottom: 20px;
}

.features-section ul {
  list-style: none;
  padding: 0;
  text-align: left;
}

.features-section li {
  font-size: 1.1rem;
  margin-bottom: 15px;
}

.features-section li strong {
  color: #99bfff;
}

/* Typing Effect */
/* Typing Effect */
.typing {
  display: inline-block;
  white-space: nowrap;
  overflow: hidden;
}

/* Typing Animation */
@keyframes typing {
  from {
    width: 0;
  }
  to {
    width: 100%;
    border: none;
  }
}

/* Blinking Cursor Animation */
@keyframes blink {
  50% {
    border-color: transparent; /* Blinks off */
  }
  100% {
    border-color: black; /* Blinks on */
  }
}