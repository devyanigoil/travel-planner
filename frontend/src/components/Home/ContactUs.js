import React from 'react';
import './ContactUs.css';
import { Modal, Button, Alert } from "react-bootstrap";

const ContactUs = () => {
  return (
    <div className="contact-us">
          <div className="contact-header">
            <div>
            <h1>Contact Us</h1>
            <p>
              Interested in working together? Fill out some info and we will be
              in touch shortly. We canâ€™t wait to hear from you!
            </p>
            </div>
          </div>
          <div className="right-div contact-section">
          <form >
            <div className="mb-3">
              <label htmlFor="firstName" className="form-label">
                Name 
              </label>
              <div className="d-flex gap-3">
                <input
                  type="text"
                  id="firstName"
                  className="form-control"
                  placeholder="First Name"
                  required
                />
                <input
                  type="text"
                  id="lastName"
                  className="form-control"
                  placeholder="Last Name"
                  required
                />
              </div>
            </div>
            <div className="mb-3">
              <label htmlFor="email" className="form-label">
                Email 
              </label>
              <input
                type="email"
                id="email"
                className="form-control"
                placeholder="Email"
                required
              />
            </div>
            <div className="mb-3">
              <label htmlFor="message" className="form-label">
                Message 
              </label>
              <textarea
                id="message"
                className="form-control"
                rows="4"
                placeholder="Your Message"
                required
              ></textarea>
            </div>
            <Button type="submit" className="custom-get-started-btn" variant="outline-primary">
            Submit
          </Button>
          </form>
          </div>
    </div>
  );
};

export default ContactUs;
