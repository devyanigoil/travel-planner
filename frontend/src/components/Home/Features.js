import React, { useEffect, useState } from "react";
import "./Features.css";
import backend from "../Utils/backend";

const Features = () => {
  const [features, setFeatures] = useState([]);

  const getFeatureList = async () => {
    try {
      const response = await backend.getFeaturesList();
      const updatedFeatures = response.data[0].features.map(
        (feature, index) => ({
          ...feature,
          id: index + 1, // Add an ID (index-based or any unique identifier logic)
        })
      );
      setFeatures(updatedFeatures);
    } catch (error) {
      console.error("Something went wrong");
    }
  };

  useEffect(() => {
    getFeatureList();
  }, []);

  return (
    <div className="features">
      <div className="left-div">
        <h2>Our Services</h2>
      </div>
      <div className="right-div">
        <div
          className="accordion custom-accordion border-0"
          id="featuresAccordion"
        >
          {features.map((feature) => (
            <div
              className="accordion-item custom-accordion-item"
              key={feature.id}
            >
              <div className="custom-accordion-header">
                <h2 className="accordion-header" id={`heading-${feature.id}`} data-testid={`heading-${feature.id}`}>
                  <button
                    className="accordion-button collapsed d-flex justify-content-between align-items-center bg-white shadow-none custom-accordion-button"
                    type="button"
                    data-bs-toggle="collapse"
                    data-bs-target={`#collapse-${feature.id}`}
                    aria-expanded="false"
                    aria-controls={`collapse-${feature.id}`}
                  >
                    <span>{feature.feature}</span>
                    <span className="toggle-icon"></span>
                  </button>
                </h2>
              </div>

              <div
                id={`collapse-${feature.id}`}
                className="accordion-collapse collapse"
                aria-labelledby={`heading-${feature.id}`}
                data-bs-parent="#featuresAccordion"
                data-testid={`collapse-${feature.id}`}
              >
                <div className="accordion-body">{feature.description}</div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Features;
