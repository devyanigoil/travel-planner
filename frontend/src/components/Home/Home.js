import React, { useState } from "react";
import "./HomePage.css";
import {
  faUserPlus,
  faHeart,
  faSearch,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Modal, Button, Alert } from "react-bootstrap";
import ContactUs from "./ContactUs";
import Features from "./Features";

const Home = () => {

  return (
    <div data-testid="home-container" className="home-container">
      <section className="header-section">
        <div class="flex-right">
          <h2 className="typing">Discover Your Next Adventure</h2>
          {/* <h3>Find Your Ideal Roommate with Ease</h3> */}
          <p>
            Welcome to your next adventure! Whether you're dreaming of a sunny
            beach escape or a mountain retreat, we're here to guide you every
            step of the way.
          </p>
          <Button className="custom-get-started-btn" variant="outline-primary">
            Get Started
          </Button>
        </div>
        <div>
          {" "}
          <img
            className="home-page-img"
            src="/images/home.png"
            alt="Loading..."
          />
        </div>
      </section>
      <Features></Features>
      <ContactUs></ContactUs>
    </div>
  );
};

export default Home;
