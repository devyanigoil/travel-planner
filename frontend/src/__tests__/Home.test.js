import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../components/Home/Home'; // Adjust the import path as needed

// Mock child components to isolate the Home component
jest.mock('../components/Home/Features', () => {
  return function DummyFeatures() {
    return <div data-testid="features-component">Mocked Features</div>;
  };
});

jest.mock('../components/Home/ContactUs', () => {
  return function DummyContactUs() {
    return <div data-testid="contact-us-component">Mocked Contact Us</div>;
  };
});

describe('Home Component', () => {
  // Render tests
  describe('Rendering', () => {
    beforeEach(() => {
      render(<Home />);
    });

    test('renders the main container', () => {
      const homeContainer = screen.getByTestId('home-container');
      expect(homeContainer).toBeInTheDocument();
      expect(homeContainer).toHaveClass('home-container');
    });

    test('renders the header section', () => {
      const headerSection = document.querySelector('.header-section');
      expect(headerSection).toBeInTheDocument();
    });

    test('displays the main headline', () => {
      const headline = screen.getByText('Discover Your Next Adventure');
      expect(headline).toBeInTheDocument();
    });

    test('renders the welcome paragraph', () => {
      const welcomeText = screen.getByText(/Welcome to your next adventure!/i);
      expect(welcomeText).toBeInTheDocument();
    });

    test('displays the Get Started button', () => {
      const getStartedButton = screen.getByRole('button', { name: /get started/i });
      expect(getStartedButton).toBeInTheDocument();
      expect(getStartedButton).toHaveClass('custom-get-started-btn');
    });

    test('renders home page image', () => {
      const homeImage = screen.getByAltText('Loading...');
      expect(homeImage).toBeInTheDocument();
      expect(homeImage).toHaveClass('home-page-img');
      expect(homeImage).toHaveAttribute('src', '/images/home.png');
    });
  });

  // Child Component Rendering
  describe('Child Components', () => {
    test('renders Features component', () => {
      render(<Home />);
      const featuresComponent = screen.getByTestId('features-component');
      expect(featuresComponent).toBeInTheDocument();
    });

    test('renders ContactUs component', () => {
      render(<Home />);
      const contactUsComponent = screen.getByTestId('contact-us-component');
      expect(contactUsComponent).toBeInTheDocument();
    });
  });

//   // Snapshot Testing
//   describe('Snapshot', () => {
//     test('matches snapshot', () => {
//       const { asFragment } = render(<Home />);
//       expect(asFragment()).toMatchSnapshot();
//     });
//   });
});