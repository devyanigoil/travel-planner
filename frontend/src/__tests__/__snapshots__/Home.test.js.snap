// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <section
      class="header-section"
    >
      <div
        class="flex-right"
      >
        <h2
          class="typing"
        >
          Discover Your Next Adventure
        </h2>
        <p>
          Welcome to your next adventure! Whether you're dreaming of a sunny beach escape or a mountain retreat, we're here to guide you every step of the way.
        </p>
        <button
          class="custom-get-started-btn btn btn-outline-primary"
          type="button"
        >
          Get Started
        </button>
      </div>
      <div>
         
        <img
          alt="Loading..."
          class="home-page-img"
          src="/images/home.png"
        />
      </div>
    </section>
    <div
      data-testid="features-component"
    >
      Mocked Features
    </div>
    <div
      data-testid="contact-us-component"
    >
      Mocked Contact Us
    </div>
  </div>
</DocumentFragment>
`;
